### PR TITLE
Add option to zip backup data by one tar file

### PR DESCRIPTION
**Pull request by sweetfriday**
@domibarton, Can you please check this PR and accept it?
I need docker image with this option for tarball.
Thanks, Stefan

### CODE REVIEW

File: init.sh [hunk 1]
```diff
@@ -7,10 +7,16 @@ set -e
 #
 
 MODE=${MODE:-BACKUP}
+TARBALL=${TARBALL:-}
  
 case "${MODE^^}" in
     'BACKUP')
-        OPTIONS=${OPTIONS:--c}
+        if [[ "${MODE^^}" == "BACKUP" ]] && [[ "${TARBALL^^}" != "" ]]
```
Comment (domibarton) [inline init.sh]: You can remove the condition `[[ "${MODE^^}" == "BACKUP" ]]` as the `MODE` is already checked in the switch/case statement.

File: init.sh [hunk 2]
```diff
@@ -112,7 +119,13 @@ then
     echo "DONE"
 
     echo "===> Starting backup..."
-    exec su -pc "mydumper ${CLI_OPTIONS}" ${USER}
+    if [[ "${TARBALL^^}" != "" ]]
+    then
+        su -pc "mydumper ${CLI_OPTIONS}" ${USER}
```
Comment (domibarton) [inline init.sh]: Correct me if I'm wrong, but running commands without `exec` is suboptimal, as you can't properly stop/kill the container… Could we probably chain the `tar` command via pipe or `&&`?

File: init.sh [hunk 3]
```diff
@@ -125,6 +138,13 @@ then
     cd ${BASE_DIR}
     echo "DONE"
 
+    if [[ "${TARBALL^^}" != "" ]]
+    then
+        RESTORE_DIR=${TARBALL}
+        rm -rf ${RESTORE_DIR}
+        su -pc "tar -xvf ${TARBALL}.gz ${RESTORE_DIR}" ${USER}
```
Comment (domibarton) [inline init.sh]: Same here, we should think about `exec` and pipes or `&&`.  
However, it's more important for the `mydumper` / `myloader` commands as they might run for a longer time than the `tar` command.

**Comment by sweetfriday** [comment]
Please check again if everything is right. @domibarton
I want to get the docker image today.
Thanks, Stefan

**Comment by domibarton** [comment]
Looks good! However, didn't test it :)

**Comment by sweetfriday** [comment]
Thanks

**Comment by sweetfriday** [comment]
@domibarton, can you please build the docker image with the latest source?

**Comment by domibarton** [comment]
It's an automated build, so it was automatically built 7 hours ago :) Can you see [this](https://hub.docker.com/r/confirm/mysql-backup/builds/bszriybc6zba57tysmb5n9d/)?

**Pull request merged**